Feature: PTooltip component

pleek91 opened the pull request:
# Description
Adds a new PTooltip component. Supports tooltip text as a prop as well as two slots to customize the tooltip itself. 

![image](https://user-images.githubusercontent.com/6200442/229231122-e31ddae5-8907-41ac-bd29-49bf07e472b4.png)

pleek91 commented on src/components/Tooltip/PTooltip.vue:
```diff
@@ -0,0 +1,88 @@
+    to?: string | Element,
+  }>()
+
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-arguments
+  const popover = ref<InstanceType<typeof PPopOver>>()
```
I have no idea why but eslint kept removing this type which would make it `Ref<any>` which isn't helpful.

brandonreid replied:
I used an a similar type and needed to instantiate it with `null`. So like `ref<InstanceType<typeof PPopOver> | null>(null)`, maybe the popover doesn't exist until some mounting event so it's not possible the const assignment is valid on instantiation?

pleek91 replied:
That did work. Thanks! Still doesn't make sense though. The ref doesn't exist until mounting but the type of a ref without a value is `ref<T | undefined>`. But least `null` gets rid of the linting issue.

stackoverfloweth reviewed (approved):
Can't believe we were able to avoid this component for as long as we did. This is top notch @pleek91. Unfortunate about the eslint thing but otherwise exactly what I expected.

pleek91 commented:
Thanks @stackoverfloweth!

pleek91 commented:
@brandonreid and @stackoverfloweth I added a couple more commits to this after looking at [ARIA: tooltip role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role) from MDN. 
- Added a 500ms delay on opening
- Added the ability to press escape to close the tooltip
- Added the ability to open the tooltip on focus (keyboard navigation)
- Added `role` and `aria-describedby` attributes to appropriate elements.
